nntrainer/layers/layer_node.cpp

```diff
@@ -16,19 +16,67 @@
 #include <nntrainer_error.h>
 #include <nntrainer_log.h>
 #include <time_dist.h>
+
+#include <base_properties.h>
+#include <common_properties.h>
 namespace nntrainer {
 
+namespace props {
+class ActivationType;
+
+/**
+ * @brief Flatten property, true if needs flatten layer afterwards
+ */
+class Flatten : public Property<bool> {
+public:
```

Comment:
Yes, this is here because the property should never be used outside of layer_node.

I wonder if we should not allow making properties for each object, but only inside `common_properties.h`?

Reply:
> I wonder if we should not allow making properties for each object, but only inside common_properties.h?

If we decide to do this, I'll do it as a separate PR. There are couple of other properties need moving :)